dbms/src/Storages/DeltaMerge/ReadThread/UnorderedInputStream.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//
+// Created by elsa on 2023/2/22.
+//
+
+#include "UnorderedInputStream.h"
+#include "Flash/Coprocessor/RuntimeFilterMgr.h"
+
+namespace DB::DM {
+
+uint64_t timeSinceEpochMillisec() {
+    using namespace std::chrono;
+    return duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
+}
+
+void UnorderedInputStream::prepareRuntimeFilter() {
+    if (runtime_filter_list.empty()) {
+        return;
+    }
+    // wait for runtime filter ready
+    auto star_wait_time = timeSinceEpochMillisec();
+    std::vector<RuntimeFilterPtr> readyRfList;
+    for (RuntimeFilterPtr rf: runtime_filter_list) {
+        bool isReady = rf->await(max_wait_time_ms - timeSinceEpochMillisec() + star_wait_time);
```

Comment:
Why do we need to use a "wait" mechanism? Can we only continue if `runtime_filter_list` is fully ready?

Reply:
We need to wait for RF ready. 
The reason is: once the ```Unorderedinputstream``` is constructed in the compile phase, it may be executed. 
But at this time, the corresponding join node may not have completed the construction of rf, and may not even start. So here's a wait.